src/software/ai/evaluation/time_to_travel.cpp

```diff
@@ -0,0 +1,70 @@
+#include <cmath>
+
+#include "software/ai/evaluation/time_to_travel.h"
+
+Duration getTimeToTravelDistance(const double distance, const double max_velocity,
+                                 const double max_acceleration,
+                                 const double initial_velocity,
+                                 const double final_velocity)
+{
+    // Bound all values to be realistic
+    double d_total = std::max(0.0, distance);
+    double v_max   = std::max(0.0, max_velocity);
+    double v_i     = std::clamp(initial_velocity, -max_velocity, max_velocity);
+    double v_f     = std::clamp(final_velocity, 0.0, max_velocity);
+    double a_max   = std::max(1e-6, max_acceleration);
+
+    // Minimum distance required to accelerate/decelerate from initial to final velocity
+    double dist_required_to_reach_v_f =
+        std::abs(std::pow(v_f, 2) - std::pow(v_i, 2)) / (2 * a_max);
+    if (dist_required_to_reach_v_f > d_total)
+    {
+        // Accelerating/decelerating instantly from initial to final velocity is not
+        // possible with in the given distance, given the robot's max acceleration,
+        // therefore the robot can not reach the desired final velocity. Calculate how
+        // long it will take for the robot to accelerate towards final velocity over
+        // distance
+        double a_max_signed = a_max;
+        if (v_f < v_i)
+        {
+            // Robot must decelerate, so it has negative acceleration
+            a_max_signed *= -1.0;
+        }
+        // t = (-Vi + sqrt(Vi^2 + 2 * a * d)) / a
+        double t_total =
+            (-v_i + std::sqrt(std::pow(v_i, 2) + 2 * a_max_signed * d_total)) /
+            a_max_signed;
+        return Duration::fromSeconds(t_total);
```

Comment:
could you double check this for the case where v_f < v_i? I'm worried that we should be looking at the solution where it's (-v_i - std::sqrt...) if the acceleration is signed as negative

Reply:
I looked abit more into it and I think just taking the positive case is okay.